### PR TITLE
Metadata panel overhaul

### PR DESCRIPTION
**Pull request by dominikl**
Overhaul of the right hand side metadata panel; respective web PR #4257; see also [Scoping of Right-Hand Pane - General Panel Display.docx](https://docs.google.com/document/d/1C__ZB0uyX3295Fop-k-fcQIL53NaKrZcuZ3N0OBDMb0/edit#)

Test: Check that the metadata panel still works as expected.

Known issues: 
- Modification of MapAnnotations always triggers the "Save data" dialog (but the data has already been saved at that stage, so approving or disapproving the save action doesn't make a difference).
- ROI panel not implemented yet.

/cc @gusferguson @jburel @pwalczysko @will-moore

### CODE REVIEW

**Comment by pwalczysko** [comment]
This PR is not in today.

**Comment by gusferguson** [comment]
@dominikl @pwalczysko @jburel 

Tested using OMERO.insight-5.2.0-m1-270-5057004-ice35-b115-win eel logged in as user-6 on own and and user-5 data read-write-1

All accordions in the General panel expand and collapse as expected.
Expanded/collapsed status is maintained moving from image to image.
All functionality within the sections appears to be working as normal.

In order to maintain coherence with Web client https://github.com/openmicroscopy/openmicroscopy/pull/4257 need to:
- leave owner under Image/Dataset/Project ID as before (Will suggested it looks better that way)
- Image/Project/Dataset Details should be open by default
- change title “Image’s details” to “Image Details” - removing ‘ and changing case of D
- similarly “Project Details” and "Dataset Details"
- all titles should be plurals: "Ratings" and “Others” are singular
- have ROIs hidden if there are no ROIs on image (understand this may not be possible to do in Insight)
-  "Others" should go after "Ratings"

Starting to look good to me.

**Comment by pwalczysko** [comment]
Select a dv, create some MAnnot, then, without saving, navigate to another image, observe error
```
java.lang.Exception: omero.gateway.exception.DSAccessException: Cannot access data. 
Cannot create the objects.
	at org.openmicroscopy.shoola.env.data.OMEROGateway.handleException(OMEROGateway.java:759)
	at org.openmicroscopy.shoola.env.data.OMEROGateway.createObjects(OMEROGateway.java:1992)
	at org.openmicroscopy.shoola.env.data.OMEROGateway.createObjects(OMEROGateway.java:1969)
	at org.openmicroscopy.shoola.env.data.OmeroMetadataServiceImpl.prepareAnnotationToAdd(OmeroMetadataServiceImpl.java:566)
	at org.openmicroscopy.shoola.env.data.OmeroMetadataServiceImpl.saveData(OmeroMetadataServiceImpl.java:1424)
	at org.openmicroscopy.shoola.env.data.views.calls.StructuredAnnotationSaver$1.doCall(StructuredAnnotationSaver.java:89)
	at org.openmicroscopy.shoola.env.data.views.BatchCall.doStep(BatchCall.java:144)
	at org.openmicroscopy.shoola.util.concur.tasks.CompositeTask.doStep(CompositeTask.java:226)
	at org.openmicroscopy.shoola.env.data.views.CompositeBatchCall.doStep(CompositeBatchCall.java:126)
	at org.openmicroscopy.shoola.util.concur.tasks.ExecCommand.exec(ExecCommand.java:165)
	at org.openmicroscopy.shoola.util.concur.tasks.ExecCommand.run(ExecCommand.java:276)
	at org.openmicroscopy.shoola.util.concur.tasks.AsyncProcessor$Runner.run(AsyncProcessor.java:91)
	at java.lang.Thread.run(Thread.java:745)
Caused by: omero.gateway.exception.DSAccessException: Cannot access data. 
Cannot update the object.
	at org.openmicroscopy.shoola.env.data.OMEROGateway.handleException(OMEROGateway.java:759)
	at org.openmicroscopy.shoola.env.data.OMEROGateway.saveAndReturnObject(OMEROGateway.java:2130)
	at org.openmicroscopy.shoola.env.data.OMEROGateway.createObjects(OMEROGateway.java:1990)
	... 11 more
Caused by: omero.ValidationException
    serverStackTrace = "ome.conditions.ValidationException: could not insert: [ome.model.annotations.MapAnnotation]; SQL [insert into annotation (description, creation_id, external_id, group_id, owner_id, permissions, update_id, name, ns, version, discriminator, id) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, '/map/', ?)]; constraint [namespace_name]; nested exception is org.hibernate.exception.ConstraintViolationException: could not insert: [ome.model.annotations.MapAnnotation]
                        	at org.springframework.orm.hibernate3.SessionFactoryUtils.convertHibernateAccessException(SessionFactoryUtils.java:637)
                        	at org.springframework.orm.hibernate3.HibernateAccessor.convertHibernateAccessException(HibernateAccessor.java:412)
                        	at org.springframework.orm.hibernate3.HibernateInterceptor.invoke(HibernateInterceptor.java:117)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:108)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at ome.tools.hibernate.ProxyCleanupFilter$Interceptor.invoke(ProxyCleanupFilter.java:249)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at ome.services.util.ServiceHandler.invoke(ServiceHandler.java:121)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
                        	at com.sun.proxy.$Proxy89.saveAndReturnArray(Unknown Source)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
                        	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                        	at java.lang.reflect.Method.invoke(Method.java:606)
                        	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
                        	at ome.security.basic.BasicSecurityWiring.invoke(BasicSecurityWiring.java:93)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at ome.services.blitz.fire.AopContextInitializer.invoke(AopContextInitializer.java:43)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
                        	at com.sun.proxy.$Proxy89.saveAndReturnArray(Unknown Source)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
                        	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                        	at java.lang.reflect.Method.invoke(Method.java:606)
                        	at ome.services.blitz.util.IceMethodInvoker.invoke(IceMethodInvoker.java:172)
                        	at ome.services.throttling.Callback.run(Callback.java:56)
                        	at ome.services.throttling.InThreadThrottlingStrategy.callInvokerOnRawArgs(InThreadThrottlingStrategy.java:56)
                        	at ome.services.blitz.impl.AbstractAmdServant.callInvokerOnRawArgs(AbstractAmdServant.java:140)
                        	at ome.services.blitz.impl.UpdateI.saveAndReturnArray_async(UpdateI.java:56)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
                        	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                        	at java.lang.reflect.Method.invoke(Method.java:606)
                        	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
                        	at omero.cmd.CallContext.invoke(CallContext.java:78)
                        	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
                        	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
                        	at com.sun.proxy.$Proxy90.saveAndReturnArray_async(Unknown Source)
                        	at omero.api._IUpdateTie.saveAndReturnArray_async(_IUpdateTie.java:80)
                        	at omero.api._IUpdateDisp.___saveAndReturnArray(_IUpdateDisp.java:220)
                        	at omero.api._IUpdateDisp.__dispatch(_IUpdateDisp.java:343)
                        	at IceInternal.Incoming.invoke(Incoming.java:222)
                        	at Ice.ConnectionI.invokeAll(ConnectionI.java:2482)
                        	at Ice.ConnectionI.dispatch(ConnectionI.java:1258)
                        	at Ice.ConnectionI.message(ConnectionI.java:1213)
                        	at IceInternal.ThreadPool.run(ThreadPool.java:321)
                        	at IceInternal.ThreadPool.access$300(ThreadPool.java:12)
                        	at IceInternal.ThreadPool$EventHandlerThread.run(ThreadPool.java:693)
                        	at java.lang.Thread.run(Thread.java:745)
                        "
    serverExceptionClass = "ome.conditions.ValidationException"
    message = "could not insert: [ome.model.annotations.MapAnnotation]; SQL [insert into annotation (description, creation_id, external_id, group_id, owner_id, permissions, update_id, name, ns, version, discriminator, id) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, '/map/', ?)]; constraint [namespace_name]; nested exception is org.hibernate.exception.ConstraintViolationException: could not insert: [ome.model.annotations.MapAnnotation]"
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at IceInternal.BasicStream.createUserException(BasicStream.java:2615)
	at IceInternal.BasicStream.access$300(BasicStream.java:12)
	at IceInternal.BasicStream$EncapsDecoder10.throwException(BasicStream.java:3099)
	at IceInternal.BasicStream.throwException(BasicStream.java:2077)
	at IceInternal.Outgoing.throwUserException(Outgoing.java:538)
	at omero.api._IUpdateDelM.saveAndReturnArray(_IUpdateDelM.java:151)
	at omero.api.IUpdatePrxHelper.saveAndReturnArray(IUpdatePrxHelper.java:451)
	at omero.api.IUpdatePrxHelper.saveAndReturnArray(IUpdatePrxHelper.java:423)
	at org.openmicroscopy.shoola.env.data.OMEROGateway.saveAndReturnObject(OMEROGateway.java:2128)
	... 12 more

	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.showErrorDialog(UserNotifierImpl.java:189)
	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.notifyError(UserNotifierImpl.java:289)
	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.notifyError(UserNotifierImpl.java:257)
	at org.openmicroscopy.shoola.agents.metadata.MetadataLoader.handleException(MetadataLoader.java:112)
	at org.openmicroscopy.shoola.agents.metadata.MetadataLoader.handleException(MetadataLoader.java:183)
	at org.openmicroscopy.shoola.env.data.events.DSCallAdapter.eventFired(DSCallAdapter.java:84)
	at org.openmicroscopy.shoola.env.data.views.BatchCallMonitor$1.run(BatchCallMonitor.java:124)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:227)
	at java.awt.Dialog.show(Dialog.java:1084)
	at java.awt.Component.show(Component.java:1654)
	at java.awt.Component.setVisible(Component.java:1606)
	at java.awt.Window.setVisible(Window.java:1014)
	at java.awt.Dialog.setVisible(Dialog.java:1005)
	at org.openmicroscopy.shoola.util.ui.UIUtilities.centerAndShow(UIUtilities.java:569)
	at org.openmicroscopy.shoola.util.ui.MessageBox.centerMsgBox(MessageBox.java:194)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerComponent.setSelectedNode(TreeViewerComponent.java:1472)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerComponent.setSelectedNode(TreeViewerComponent.java:1633)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerControl.propertyChange(TreeViewerControl.java:1180)
	at org.openmicroscopy.shoola.util.ui.component.AbstractComponent.firePropertyChange(AbstractComponent.java:147)
	at org.openmicroscopy.shoola.agents.dataBrowser.view.DataBrowserComponent.setSelectedDisplay(DataBrowserComponent.java:431)
	at org.openmicroscopy.shoola.agents.dataBrowser.view.DataBrowserControl.propertyChange(DataBrowserControl.java:495)
	at org.openmicroscopy.shoola.util.ui.component.AbstractComponent.firePropertyChange(AbstractComponent.java:147)
	at org.openmicroscopy.shoola.agents.dataBrowser.browser.BrowserModel.setSelectedDisplay(BrowserModel.java:822)
	at org.openmicroscopy.shoola.agents.dataBrowser.browser.BrowserControl.handleSelection(BrowserControl.java:268)
	at org.openmicroscopy.shoola.agents.dataBrowser.browser.BrowserControl.onClick(BrowserControl.java:187)
	at org.openmicroscopy.shoola.agents.dataBrowser.browser.BrowserControl.mousePressed(BrowserControl.java:600)
	at java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:280)
	at java.awt.Component.processMouseEvent(Component.java:6522)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6290)
	at java.awt.Container.processEvent(Container.java:2234)
	at java.awt.Component.dispatchEventImpl(Component.java:4881)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4530)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
	at java.awt.Container.dispatchEventImpl(Container.java:2278)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by dominikl** [comment]
Hiding the ROI taskpane might lead to a strange visual effect. The metadata panel is build in the moment you click on an object (no information about the ROIs then), then the data loading is initiated. Either hiding it from the start and just showing it after the data arrives or the other way round, will have the effect that something will 'pop up' or 'disappear' from the right hand side panel, might look a bit strange.

**Comment by gusferguson** [comment]
@dominikl  - good point - go with your recommendation.

**Comment by pwalczysko** [comment]
MAnnotations are displayed doubly each time when I create them (user-6 in rw group working on user-5 data)

screen shot 2015-10-16 at 12 05 51

**Comment by pwalczysko** [comment]
The way to repeat   https://github.com/openmicroscopy/openmicroscopy/pull/4265#issuecomment-148681991  is : 
  - find image ID: 9608 user-6 on eel merge
  - add new rows to the MAnnot and write into the fields
  - without saving, select the image (next to the left of it) ID: 9604
  - observe error

**Comment by pwalczysko** [comment]
Other than the problems with MAnnot https://github.com/openmicroscopy/openmicroscopy/pull/4265#issuecomment-148685150  https://github.com/openmicroscopy/openmicroscopy/pull/4265#issuecomment-148681991   all is fine

**Comment by sbesson** [comment]
See Javadoc warnings in https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-build/121/warnings39Result/

File: components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+        c.gridy++;
+    }
+
+    /**
+     * Show only specific types of MapAnnotations
+     * 
+     * @param filter
+     *            One of: {@link Filter#SHOW_ALL}, see
+     *            {@link Filter#ADDED_BY_ME}, see
+     *            {@link Filter#ADDED_BY_OTHERS}
```
Comment (sbesson) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java]: From https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/125/consoleTest

```
09:15:54 Building tree for all the packages and classes...

09:15:59 /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java:337: warning - Tag @link: reference not found: Filter#SHOW_ALL
09:15:59 /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java:337: warning - Tag @link: reference not found: Filter#ADDED_BY_ME
09:15:59 /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java:337: warning - Tag @link: reference not found: Filter#ADDED_BY_OTHERS
```

**Comment by jburel** [comment]
this PR and gh-4257 must be in synch before we merge

File: components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java (hunk 1 above)
Comment (jburel) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/metadata/editor/MapTaskPaneUI.java]: PR opened against this branch

**Comment by pwalczysko** [comment]
Desync issue noticed between Insight and Web: 

  -  navigate to https://eel.openmicroscopy.org/merge/webclient/?show=image-9532 in Web, user-2, and go Acqiuisition pane > FITC
  - navigate to the same image in Insight (mege build)
  - observe that the FITC tab is missing (not expected)
  - the FITC tab contains some timelapse info which used to be present in the recent past on this image even in Insight (see https://github.com/openmicroscopy/openmicroscopy/pull/4199)

--> Insight to adjust

**Comment by pwalczysko** [comment]
Re: https://github.com/openmicroscopy/openmicroscopy/pull/4265#issuecomment-149908858 - confirmed that the Insight from ``latest`` build when viewing the same image, *has* the FITC tab as expected. 


screen shot 2015-10-21 at 15 22 50

**Comment by jburel** [comment]
@pwalczysko currently looking at the image on eel/merge.
No problem at all I can see FITC

**Comment by pwalczysko** [comment]
@jburel : Me too. Cannot reproduce. Tried several times. Probably okay.

**Comment by dominikl** [comment]
Sorry, saw your PR too late to go into today's build @jburel . But shouldn't affect any testing, as it's a javadoc commit only. I will check later what else is not sync with Web (note: this PR is only about the *General* metadata panel). What should actually happen with the *ROIs* tab which is supposed to provide a detailed list about the ROIs (in Insight it's just an empty panel as 'placeholer' at the moment), probably not for 5.2.0, right?

**Comment by jburel** [comment]
@dominikl: it is currently a placeholder. so we can start adding element during 5.2.x

**Comment by jburel** [comment]
Discussed today with @pwalczysko and @will-moore:
the tab "other" should only be added when we have annotation of type "other"
It should be added at the bottom so it does not disrupt the users
cc @gusferguson

**Comment by dominikl** [comment]
Tests for last commit:
- Select an image with no "other" annotations: The "Others" task pane should not be displayed.
- Select an image with "other" annotations (e. g. user-3 24103 ): The "Others" task pane should be displayed as last task pane in the General tab.
- There "ROIs" task pane shouldn't be displayed at all (because it's not implemented yet).

**Comment by jburel** [comment]
The tabs are correctly organized
The layout of components within each time will have to be reviewed in 5.2.1 e.g. rating
Currently Add comment does not work. Need to click on save in order to save the comments
Also when I start typing the text area becomes much bigger.

**Comment by jburel** [comment]
in AnnotationTaskPane.java typo ```ahve```

**Comment by jburel** [comment]
in ```CommentsTaskPaneUI```: 
```
/** The collection of annotation to display. */
private List annotationToDisplay;
````
should be annotations

**Comment by jburel** [comment]
Add comment now works as expected.
A minor point (to be fixed in another PR):
 * if you add comment, the comment is saved but the Save icon in the toolbar is active. This has no effect when the switching to another object i.e. no dialog pops up asking "Do you want to save?"
note the button is enabled only when adding a comment on image.

**Comment by jburel** [comment]
Comment added to https://trello.com/c/74yV81oZ/9-insight-right-hand-panel-follow-up

**Pull request merged**